it/controllers/pensions/incomeFromPensions/StatePensionAmountControllerISpec.scala

```diff
@@ -0,0 +1,494 @@
+          h1Check(user.specificExpectedResults.get.expectedHeading)
+          captionCheck(expectedCaption(taxYearEOY), captionSelector)
+          textOnPageCheck(hintText, hintTextSelector)
+          textOnPageCheck(poundPrefixText, poundPrefixSelector)
+          textOnPageCheck(expectedWhereToFindThisInformation, whereToFindThisInformationSelector)
+          textOnPageCheck(expectedYouCanFindThisInformationIn, paragraphFindThisInfoSelector)
+          textOnPageCheck(user.specificExpectedResults.get.expectedDetailsExample1, paragraphSelector(1))
+          textOnPageCheck(user.specificExpectedResults.get.expectedDetailsExample2, paragraphSelector(2))
+          inputFieldValueCheck(amountInputName, inputSelector, "")
+          buttonCheck(buttonText, continueButtonSelector)
+          formPostLinkCheck(statePensionAmountUrl(taxYearEOY), formSelector)
+          welshToggleCheck(user.isWelsh)
+        }
+
+        "render Did you get a state pension amount page page prefilled when cya data" which {
+
+          val existingAmount: String = "999.88"
+          lazy val result: WSResponse = {
+            dropPensionsDB()
+            authoriseAgentOrIndividual(user.isAgent)
+            val pensionsViewModel = anIncomeFromPensionsViewModel.copy(
+              statePension = Some(anStateBenefitViewModelOne.copy(amountPaidQuestion = Some(true), amount = Some(BigDecimal(existingAmount)))))
+            insertCyaData(pensionsUsersData(isPrior = false, aPensionsCYAModel.copy(incomeFromPensions = pensionsViewModel)), aUserRequest)
+            urlGet(fullUrl(statePensionAmountUrl(taxYearEOY)), user.isWelsh, follow = false,
+              headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY)))
+          }
+
+          implicit def document: () => Document = () => Jsoup.parse(result.body)
+
+          "has an OK status" in {
+            result.status shouldBe OK
+          }
+          titleCheck(user.specificExpectedResults.get.expectedTitle)
+          h1Check(user.specificExpectedResults.get.expectedHeading)
+          captionCheck(expectedCaption(taxYearEOY), captionSelector)
+          textOnPageCheck(poundPrefixText, poundPrefixSelector)
+          textOnPageCheck(hintText, hintTextSelector)
+          textOnPageCheck(expectedWhereToFindThisInformation, whereToFindThisInformationSelector)
+          textOnPageCheck(expectedYouCanFindThisInformationIn, paragraphFindThisInfoSelector)
+          textOnPageCheck(user.specificExpectedResults.get.expectedDetailsExample1, paragraphSelector(1))
+          textOnPageCheck(user.specificExpectedResults.get.expectedDetailsExample2, paragraphSelector(2))
+          inputFieldValueCheck(amountInputName, inputSelector, existingAmount)
+          buttonCheck(buttonText, continueButtonSelector)
+          formPostLinkCheck(statePensionAmountUrl(taxYearEOY), formSelector)
+          welshToggleCheck(user.isWelsh)
+        }
+      }
+    }
+
+    "the user is in year" which {
```

Comment:
Could you specify this test a bit more, e.g. redirect to overview page if... and in the submit one too?